### PR TITLE
Add support for node::Encode/DecodeBytes/DecodeWrite.

### LINKED ISSUE #250: Add support for node::Encode/DecodeBytes/DecodeWrite
Could you add support for those handy utils?

```cpp
#if NODE_VERSION_AT_LEAST(0,11,12)
#define NanEncode(buf, buflen, encoding) \
  node::Encode(v8::Isolate::GetCurrent(), buf, buflen, encoding)
#define NanDecodeBytes(val, encoding) \
  node::DecodeBytes(v8::Isolate::GetCurrent(), val, encoding)
#define NanDecodeWrite(buf, buflen, val, encoding) \
  node::DecodeWrite(v8::Isolate::GetCurrent(), buf, buflen, val, encoding)
#else
#define NanEncode(buf, buflen, encoding) \
  node::Encode(buf, buflen, encoding)
#define NanDecodeBytes(val, encoding) \
  node::DecodeBytes(val, encoding)
#define NanDecodeWrite(buf, buflen, val, encoding) \
  node::DecodeWrite(buf, buflen, val, encoding)
#endif
```

### PR DESCRIPTION
**Pull request by kkoopa**
Fixes #250.

### CODE REVIEW

File: nan.h [hunk 1]
```diff
@@ -1816,6 +1816,45 @@ namespace Nan {
   enum Encoding {ASCII, UTF8, BASE64, UCS2, BINARY, HEX, BUFFER};
 }
 
+NAN_INLINE v8::Local<v8::Value> NanEncode(const void *buf, size_t len, enum Nan::Encoding encoding = Nan::BINARY) {
```
Comment (bnoordhuis) [inline nan.h]: Maybe line-wrap this?

File: nan.h [hunk 2]
```diff
@@ -1816,6 +1816,45 @@ namespace Nan {
...
+#if (NODE_MODULE_VERSION > 0x000B)
+return node::Encode(v8::Isolate::GetCurrent(), buf, len, static_cast<node::encoding>(encoding));
+#else
+# if  (NODE_MODULE_VERSION < 0x000B)
+  if (encoding == Nan::BUFFER) {
+    return node::Encode(buf, len, node::BINARY);
```
Comment (bnoordhuis) [inline nan.h]: This should be `return node::Buffer::New(buf, len);`.

File: nan.h [hunk 3]
```diff
@@ -1816,6 +1816,45 @@ namespace Nan {
...
+#endif
+}
+
+NAN_INLINE ssize_t NanDecodeBytes(v8::Handle<v8::Value> val, enum Nan::Encoding encoding = Nan::BINARY) {
+#if (NODE_MODULE_VERSION > 0x000B)
+  return node::DecodeBytes(v8::Isolate::GetCurrent(), val, static_cast<node::encoding>(encoding));
+#else
+# if (NODE_MODULE_VERSION < 0x000B)
+  if (encoding == Nan::BUFFER) {
+    return node::DecodeBytes(val, node::BINARY);
```
Comment (bnoordhuis) [inline nan.h]: `DecodeBytes(buf, BUFFER)` calls ToString() on the buffer, which interprets (for better or worse) the contents of the buffer as UTF-8.  It's arguably a bug but it's better to be compatible.

**Comment by bnoordhuis** [comment]
The DecodeBytes() and DecodeWrite() calls that transpose BUFFER to BINARY work but perhaps it's clearer to call the corresponding Buffer methods directly.

**Comment by kkoopa** [comment]
I did that first, but changed to this after I noticed the code paths would become identical as it felt cleaner and does not perform the same sanity checks twice. Otherwise I need to replicate a bunch of error checking (val is not, array, because 'raw' encoding is no longer supported) and other edge cases, which makes it grow more than one might first think.

File: nan.h (hunk 1 above)
Comment (kkoopa) [inline nan.h]: Ah, yes. I forgot. Mostly wanted to see that Travis did not choke.

**Comment by kkoopa** [comment]
Should be fixed now.

**Pull request closed without merging**